source/agora/consensus/validation/Block.d

```diff
@@ -639,14 +647,17 @@ version (unittest)
     public string isValidcheck (const ref Block block, Height prev_height,
         Hash prev_hash, scope UTXOFinder findUTXO,
         size_t active_enrollments, size_t enrolled_validators, scope FeeChecker checkFee,
-        scope EnrollmentFinder findEnrollment, ulong enrollment_cycle = 0,
-        ulong prev_timestamp = 0, ulong curr_timestamp = ulong.max,
+        scope EnrollmentFinder findEnrollment, Hash random_seed = Hash.init,
+        ulong enrollment_cycle = 0, ulong prev_timestamp = 0, ulong curr_timestamp = ulong.max,
         Duration block_timestamp_tolerance_dur = 100.seconds,
         string file = __FILE__, size_t line = __LINE__) nothrow @safe
     {
+        if (random_seed == Hash.init)
+            random_seed = getTestRandomSeed();
```

Comment:
Why not just give it as default parameter ?

Reply:
> Why not just give it as default parameter ?

It's because there are some cases the `random_seed` is already set before calling the `isValidcheck` function. In many test cases, there are two situations where the `random_seed` is set or not.

Reply:
But if it's set, it can be passed as a parameter ? Maybe you have a link to the code that creates the issue ?

Reply:
> But if it's set, it can be passed as a parameter? Maybe you have a link to the code that creates the issue?

I'll make the value be passed as a parameter. I think it's the last thing to do in this PR.